sdk/cosmosdb/cosmos/test/internal/session.spec.ts

```diff
@@ -119,13 +121,31 @@ describe.skip("Session Token", function (this: Suite) {
     // Create items using both clients so they each establish a session with the backend
     const container2 = clientB.database(dbId).container(containerId);
     await Promise.all([createItem(container), createItem(container2)]);
+  });
+
+  it("Should check if the MaxIntegratedCacheStaleness header is set.", async function () {
+    const dbId = addEntropy("maxIntegratedCacheTestDB");
+    const containerId = addEntropy("maxIntegratedCacheTestContainer");
+
+    const client = new CosmosClient({
+      endpoint,
+      key: masterKey,
+      consistencyLevel: ConsistencyLevel.Eventual,
+    });
+
+    // Create Database and Container
+    const { database } = await client.databases.createIfNotExists({
+      id: dbId,
+    });
+    const { container } = await database.containers.createIfNotExists({
+      id: containerId,
+    });
+    client.database(dbId).container(containerId);
+    const spy = sinon.spy(container, "read");
 
-    // Create an item using client
-    const id = await createItem(container);
-    const { resource, statusCode } = await container2.item(id).read();
-    console.log(statusCode, resource);
-    assert.ok(resource);
-    assert.strictEqual(statusCode, 200);
+    // Container read with consistency level and max integrated cache staleness(milliseconds)
+    await container.read({ maxIntegratedCacheStaleness: 2001 });
+    spy.getCall(0).calledWith({ maxIntegratedCacheStaleness: 0 });
```

Comment:
{ maxIntegratedCacheStaleness: 0 }, isn;t this different from original passed arguents?

Reply:
It checks if he property exist not the value.

Reply:
Isn't it a requirement/goal to validate the value?

Reply:
fixed